### PR TITLE
A successful Rails 6.0 upgrade

### PR DESCRIPTION
**Pull request by j-maynard**
OK some caveats...  I have turned off sentry in development/test.  I
have done this because it was masking errors locally as it was uanble
to connect to an instance of sentry.

Rspec 3.8 is not compatible with Rails 6 and technically Rails 4.0
is not out yet.  So to make Rspec work I have had to upgrade the Gem
to the beta version.  I think this will stop us upgrading to Rails 6.0
at the moment.  I haven't been able to find an expected release date
for Rspec 4.0 so its a choice to having a beta package in our app
(even if it is for testing) or to stick with Rails 5.2.3.

I have raised this PR and opened it up for comments from people.

### CODE REVIEW

File: config/storage.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+test:
```
Comment (jakubmiarka) [inline config/storage.yml]: why are we adding the storage and updating the DB schema?

File: Gemfile [hunk 2]
```diff
@@ -37,40 +39,44 @@ gem 'colorize'
...
   gem 'listen', '>= 3.0.5', '< 3.2'
+  gem 'web-console', '>= 3.3.0'
   # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
   gem 'spring'
   gem 'spring-watcher-listen', '~> 2.0.0'
 end
 
-gem 'dotenv-rails', require: 'dotenv/rails-now', group: [:development, :test]
+group :production do
+  gem 'sentry-raven'
```
Comment (jakubmiarka) [inline Gemfile]: I don't think the environment is set as `production` during the build stage (when `bundle install` runs). So we either need to update the command or move this back to the default group. Is it error-crashing out in Rails 6? It always gave warning that a DSN is not set in the logs (locally).

File: config/storage.yml (hunk 1 above)
Comment (j-maynard) [reply config/storage.yml]: I have removed the active storage stuff.

File: Gemfile (hunk 2 above)
Comment (j-maynard) [reply Gemfile]: I have moved the gem back to the default but kept the code which limits sentry to only run in production.

File: db/schema.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2019_10_07_201138) do
+ActiveRecord::Schema.define(version: 2019_08_31_100615) do
```
Comment (jakubmiarka) [inline db/schema.rb]: this should not change...

File: config/initializers/sentry.rb [hunk 4]
```diff
@@ -1,4 +1,6 @@
-Raven.configure do |config|
-  config.sanitize_fields = Rails.application.config.filter_parameters.map(&:to_s)
-  config.transport_failure_callback = ->(event) { Rails.logger.error(event.to_s) }
-end
+if Rails.env.production? 
+  Raven.configure do |config|
+    config.sanitize_fields = Rails.application.config.filter_parameters.map(&:to_s)
+    config.transport_failure_callback = ->(event) { Rails.logger.error(event.to_s) }
+  end
+end
```
Comment (brenetic) [inline config/initializers/sentry.rb]:
```suggestion
Raven.configure do |config|
  config.sanitize_fields = Rails.application.config.filter_parameters.map(&:to_s)
  config.transport_failure_callback = ->(event) { Rails.logger.error(event.to_s) }
  config.environments = %w(production)
end	    
```

So from what I remember the errors are just that Raven hasn't got a DSN set so can't send the errors. If we don't want this noisy output in development I _think_ it's possible to just set `config.environments = %w(production)` as above.

Comment (j-maynard) [reply config/initializers/sentry.rb]: Thank you for this... this solved the issues of log messages not coming through.

**Pull request merged**